beta-pipelines/skylab/m3c/CondensedSnm3C.wdl

```diff
@@ -569,72 +570,101 @@ task Hisat_single_end_r1_r2_mapping_dna_mode_and_merge_sort_split_reads_by_name
-       Int mem_size = 20  
-   }
-
-   command <<<
-       set -euo pipefail
-       # unzip bam file
-       tar -xf ~{bam}
-       rm ~{bam}
-
-       # create output dir
-       mkdir /cromwell_root/output_bams
-
-       # get bams
-       bams=($(ls | grep "sort.bam$"))
-
-       # loop through bams and run python script on each bam 
-       # scatter instead of for loop to optimize
-       python3 <<CODE
-       from cemba_data.hisat3n import *
-       import os
-       bams="${bams[@]}"
-       for bam in bams.split(" "):
+    input {
+        File bam
+        String docker = "us.gcr.io/broad-gotc-prod/m3c-yap-hisat:1.0.0-2.2.1"
+        Int disk_size = 80
+        Int mem_size = 20  
+    }
+
+    command <<<
+        set -euo pipefail
+        # unzip bam file
+        tar -xf ~{bam}
+        rm ~{bam}
+
+        # create output dir
+        mkdir /cromwell_root/output_bams
+
+        # get bams
+        bams=($(ls | grep "sort.bam$"))
+
+        # loop through bams and run python script on each bam 
+        # scatter instead of for loop to optimize
+        python3 <<CODE
+        from cemba_data.hisat3n import *
+        import os
+        bams="${bams[@]}"
+        for bam in bams.split(" "):
             name=".".join(bam.split(".")[:3])+".read_overlap.bam"
             remove_overlap_read_parts(in_bam_path=os.path.join(os.path.sep, "cromwell_root", bam), out_bam_path=os.path.join(os.path.sep, "cromwell_root", "output_bams", name))
-       CODE
+        CODE
        
-       cd /cromwell_root/output_bams
+        cd /cromwell_root/output_bams
        
-       #tar up the merged bam files
-       tar -zcvf ../remove_overlap_read_parts.tar.gz *bam
+        #tar up the merged bam files
+        tar -zcvf ../remove_overlap_read_parts.tar.gz *bam
        
-   >>>
-   runtime {
-       docker: docker
-       disks: "local-disk ${disk_size} HDD"
-       cpu: 1
-       memory: "${mem_size} GiB"
-   }
-   output {
+    >>>
+    runtime {
+        docker: docker
+        disks: "local-disk ${disk_size} HDD"
+        cpu: 1
+        memory: "${mem_size} GiB"
+    }
+    output {
         File output_bam_tar = "remove_overlap_read_parts.tar.gz"
     }
-
 }
 
-#task merge_original_and_split_bam_and_sort_all_reads_by_name_and_position {
-#    input {
-#        File unique_bam
-#        File split_bam
-#    }
-#    command <<<
-#    >>>
-#    runtime {
-#        docker: "fill_in"
-#        disks: "local-disk ${disk_size} HDD"
-#        cpu: 1
-#        memory: "${mem_size} GiB"
-#    }
-#    output {
-#        File name_sorted_bam = ""
-#        File position_sorted_bam = ""
-#    }
-#}
+task merge_original_and_split_bam_and_sort_all_reads_by_name_and_position {
+    input {
+        File bam
+        File split_bam
+        String docker = "us.gcr.io/broad-gotc-prod/m3c-yap-hisat:1.0.0-2.2.1"
+        Int disk_size = 80
+        Int mem_size = 20
+    }
+    command <<<
+      set -euo pipefail
+      #unzip bam file
+      pwd
```

Comment:
We could probably clean up these pwd and ls statements!

Reply:
Thank you! I cleaned up a little and for the ones I left in because I thought it might be helpful for logging, I added echo statements to make it clearer (hopefully).